src/tf/tf.cpp

```diff
@@ -939,23 +1033,43 @@ struct tf_parser
                     continue;
                 if(nodes.count(input) > 0)
                 {
-                    auto&& iname = get_name(nodes.at(input));
-                    assert(name != iname);
-                    this->parse_node(iname);
-                    args.push_back(instructions.at(iname));
+                    std::string iname;
+                    // input was from a node with multiple outputs
+                    if(contains(input, ':'))
+                    {
+                        auto orig_iname = input.substr(0, input.find(':'));
+                        assert(name != orig_iname);
+                        this->parse_node(orig_iname);
+                        args.push_back(instructions.at(input));
+                    }
+                    else
+                    {
+                        iname = get_name(nodes.at(input));
+                        assert(name != iname);
+                        this->parse_node(iname);
+                        args.push_back(instructions.at(iname));
+                    }
                 }
                 else
                 {
                     args.push_back(instructions.at(input));
                 }
             }
+
+            std::vector<instruction_ref> result;
             if(ops.count(node.op()) == 0)
             {
-                instructions[name] = prog.add_instruction(op::unknown{node.op()}, args);
+                result.push_back(prog.add_instruction(op::unknown{node.op()}, args));
             }
             else
             {
-                instructions[name] = ops[node.op()](get_attributes(node), args);
+                result = ops[node.op()](get_attributes(node), args);
+            }
+            // First output has no ":" delimiter
+            instructions[name] = result.front();
```

Comment:
We should check that `result` is not empty, maybe with an `assert`.

Reply:
fixed